pkg/sink/codec/debezium/codec.go

```diff
@@ -0,0 +1,408 @@
+			if err != nil {
+				return nil, cerror.WrapError(
+					cerror.ErrDebeziumEncodeFailed,
+					err)
+			}
+			return d.Microseconds(), nil
+		}
+		return nil, cerror.WrapError(
+			cerror.ErrDebeziumEncodeFailed,
+			errors.Errorf(
+				"unexpected column value type %T for time column %s",
+				col.Value,
+				col.Name))
+	case mysql.TypeLonglong:
+		if col.Flag.IsUnsigned() {
+			// Handle with BIGINT UNSIGNED.
+			// Debezium always produce INT64 instead of UINT64 for BIGINT.
+			if v, ok := col.Value.(uint64); ok {
+				return int64(v), nil
+			}
+			return nil, cerror.WrapError(
+				cerror.ErrDebeziumEncodeFailed,
+				errors.Errorf(
+					"unexpected column value type %T for unsigned bigint column %s",
+					col.Value,
+					col.Name))
+		}
+
+		// Note: Although Debezium's doc claims to use INT32 for INT, but it
+		// actually uses INT64. Debezium also uses INT32 for SMALLINT.
+		// So we only handle with TypeLonglong here.
+	}
+	return col.Value, nil
+}
+
+func (c *Codec) rowChangeToDebeziumMsg(e *model.RowChangedEvent) (*debeziumDataChangeMsg, error) {
+	commitTime, _ := tsoutil.ParseTS(e.CommitTs)
+
+	source := &debeziumMsgSource{
+		Version:   "2.4.0.Final",
+		Connector: "TiCDC",
+		Name:      c.clusterID,
+		TsMs:      commitTime.UnixMilli(),
+		Snapshot:  false,
+		Db:        e.Table.Schema,
+		Table:     e.Table.Table,
+		ServerID:  0,
+		GtID:      nil,
+		File:      "",
+		Pos:       0,
+		Row:       0,
+		Thread:    0,
+		Query:     nil,
+
+		CommitTs:  e.CommitTs,
+		ClusterID: c.clusterID,
+	}
+
+	payload := &debeziumDataChangeMsgPayload{
+		Source:      source,
+		TsMs:        c.nowFunc().UnixMilli(),
+		Transaction: nil,
+	}
+
+	applyBefore := func() error {
+		payload.Before = make(map[string]any)
+		for i, col := range e.PreColumns {
+			value, err := c.convertToDebeziumField(col, e.ColInfos[i].Ft)
+			if err != nil {
+				return err
+			}
+			payload.Before[col.Name] = value
+		}
+		return nil
+	}
+
+	applyAfter := func() error {
+		payload.After = make(map[string]any)
+		for i, col := range e.Columns {
+			value, err := c.convertToDebeziumField(col, e.ColInfos[i].Ft)
+			if err != nil {
+				return err
+			}
+			payload.After[col.Name] = value
+		}
+		return nil
+	}
+
+	if e.IsInsert() {
+		payload.Op = "c"
+		payload.Before = nil
+		if err := applyAfter(); err != nil {
+			return nil, err
+		}
+	} else if e.IsDelete() {
+		payload.Op = "d"
+		payload.After = nil
+		if err := applyBefore(); err != nil {
+			return nil, err
+		}
+	} else if e.IsUpdate() {
+		payload.Op = "u"
+		if err := applyBefore(); err != nil {
+			return nil, err
+		}
+		if err := applyAfter(); err != nil {
+			return nil, err
+		}
+	}
+
+	return &debeziumDataChangeMsg{
+		Payload: payload,
+	}, nil
+}
+
+func (c *Codec) EncodeRowChangedEvent(
+	e *model.RowChangedEvent,
+) ([]byte, error) {
+	m, err := c.rowChangeToDebeziumMsg(e)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return json.Marshal(m)
```

Comment:
for any error comes from third party library, wrap it into TiCDC RFC error.

just like `cerror.WrapError(cerror.ErrEncodeFailed, err)`